src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int currentCapacity;
+    private int currentIndex;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void add(T value) {
-
+        addElementToArray(value, currentIndex);
```

Comment:
do you need to pass currentIndex as parameter?

Reply:
Yes, just for one example when I need to add a new element somewhere in-between of existing elements. In the rest of cases I don't need it, but I think no point to create another method addElementToArray with one parameter, right?

Reply:
Just to clarify - I'm using addElementToArray in two cases. And in the second call (of this method) I need to pass one variable (index) from that method. 
On the 22 line (to which you pointed out) I don't really need to pass currrentIndex but because method expect "index" I have to stipulate it.